src/freenet/node/OpennetManager.java

```diff
@@ -546,6 +589,39 @@ public boolean wantPeer(OpennetPeerNode nodeToAddNow, boolean addAtLRU, boolean
 				return false;
 			}
 		}
+		// Don't connect to peers in excluded countries
+		if (nodeToAddNow != null) {
+			Peer[] handshakeIPs = nodeToAddNow.getHandshakeIPs();
+			if (handshakeIPs != null) {
+				for (Peer peer : handshakeIPs) {
+					if (peer == null) {
+						continue;
+					}
+					FreenetInetAddress address = peer.getFreenetAddress();
+					if (address == null) {
+						continue;
+					}
+					InetAddress inetAddress = address.getAddress(false);
+
+					IPConverter ipConverter = IPConverter.getInstance(NodeFile.IPv4ToCountry.getFile(node));
+					try {
+						Country country = ipConverter.locateIP(inetAddress.getAddress());
+						if (country != null) {
+							if (excludedCountries.contains(country.toString().toUpperCase())) {
```

Comment:
This is missing thread synchronization against concurrent modification of excludedCountries, i.e. the user changing the configuration meanwhile.

Quote from https://docs.oracle.com/javase/7/docs/api/java/util/HashSet.html:
> Note that this implementation is not synchronized.